protocol_decoder/src/decoding.rs

```diff
@@ -268,7 +283,9 @@ impl ProcessedBlockTrace {
         gen_inputs: &mut Vec<TxnProofGenIR>,
         other_data: &OtherBlockData,
         initial_trie_state: &PartialTrieState,
-    ) {
+    ) -> bool {
+        let mut dummies_added = true;
+
```

Comment:
nit: I think it'd be more "Rust-like" to add a comment explaining what the output boolean is, so that we can directly have 

```rust
match gen_inputs.len() {
    0 => {
        do_something();
        true
    }
    _ => false,
}
```
that returns directly (no intermediary variable)